examples/pennylane_run_quantum_natural_gradient.py

```diff
@@ -0,0 +1,451 @@
+`(Harrow and Napp, 2019) <https://arxiv.org/abs/1901.05374>`__.
+The space of quantum states instead possesses a unique invariant metric
+tensor known as the Fubini-Study metric tensor :math:`g_{ij}`, which can be used to
+construct a quantum analog to natural gradient descent:
+
+.. math:: \theta_{t+1} = \theta_t - \eta g^{+}(\theta_t)\nabla \mathcal{L}(\theta),
+
+where :math:`g^{+}` refers to the pseudo-inverse.
+
+.. note::
+
+    It can be shown that the Fubini-Study metric tensor reduces
+    to the Fisher information matrix in the classical limit.
+
+    Furthermore, in the limit where :math:`\eta\rightarrow 0`,
+    the dynamics of the system are equivalent to imaginary-time
+    evolution within the variational subspace, as proposed in
+    `McArdle et al. (2018) <https://arxiv.org/abs/1804.03023>`__.
+"""
+
+##############################################################################
+# Block-diagonal metric tensor
+# ----------------------------
+#
+# A block-diagonal approximation to the Fubini-Study metric tensor
+# of a variational quantum circuit can be evaluated on quantum hardware.
+#
+# Consider a variational quantum circuit
+#
+# .. math::
+#
+#     U(\mathbf{\theta})|\psi_0\rangle = V_L(\theta_L) W_L V_{L-1}(\theta_{L-1}) W_{L-1}
+#       \cdots V_{\ell}(\theta_{\ell}) W_{\ell} \cdots V_{0}(\theta_{0}) W_{0} |\psi_0\rangle
+#
+# where
+#
+# * :math:`|\psi_0\rangle` is the initial state,
+# * :math:`W_\ell` are layers of non-parametrized quantum gates,
+# * :math:`V_\ell(\theta_\ell)` are layers of parametrized quantum gates
+#   with :math:`n_\ell` parameters :math:`\theta_\ell = \{\theta^{(\ell)}_0, \dots, \theta^{(\ell)}_n\}`.
+#
+# Further, assume all parametrized gates can be written in the form
+# :math:`X(\theta^{(\ell)}_{i}) = e^{i\theta^{(\ell)}_{i} K_i}`,
+# where :math:`K^{(\ell)}_i` is the *generator* of the parametrized operation.
+#
+# For each parametric layer :math:`\ell` in the variational quantum circuit
+# the :math:`n_\ell\times n_\ell` block-diagonal submatrix
+# of the Fubini-Study tensor :math:`g_{ij}^{(\ell)}` is calculated by:
+#
+# .. math::
+#
+#     g_{ij}^{(\ell)} = \langle \psi_{\ell-1} | K_i K_j | \psi_{\ell-1} \rangle
+#     - \langle \psi_{\ell-1} | K_i | \psi_{\ell-1}\rangle
+#     \langle \psi_{\ell-1} |K_j | \psi_{\ell-1}\rangle
+#
+# where
+#
+# .. math::
+#
+#     | \psi_{\ell-1}\rangle = V_{\ell-1}(\theta_{\ell-1}) W_{\ell-1} \cdots V_{0}(\theta_{0}) W_{0} |\psi_0\rangle.
+#
+# (that is, :math:`|\psi_{\ell-1}\rangle` is the quantum state prior to the application
+# of parameterized layer :math:`\ell`), and we have :math:`K_i \equiv K_i^{(\ell)}` for brevity.
+#
+# Let's consider a small variational quantum circuit example coded in PennyLane:
+
+import numpy as np
+
+import pennylane as qml
+from pennylane import expval, var
+
+dev = qml.device('default.qubit', wires=3)
+
+@qml.qnode(dev)
+def circuit(params):
+    # non-parametrized gates

```

Comment:
to avoid confusion, we should make it clear that these are to prepare state psi_0

Reply:
Done!